drivers/sensor/lis2mdl/Kconfig

```diff
@@ -0,0 +1,123 @@
+
+# Copyright (c) 2018 STMicroelectronics
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig LIS2MDL
+	bool
+	prompt "LIS2MDL Magnetometer"
+	depends on I2C
+	default n
+	help
+	  Enable driver for LIS2MDL I2C-based magnetometer sensor.
+
+config LIS2MDL_NAME
```

Comment:
CONFIG_LIS2MDL_I2C_MASTER_DEV_NAME, CONFIG_LIS2MDL_I2C_ADDR and LIS2MDL_I2C_MASTER_DEV_NAME will be generated from device tree "compilation".
Hence in order to prevent having them genetared twice, you should protect them under ```if !HAS_DTS_I2C_DEVICE``` directive.
You can check other drivers like LIS3MDL to check how this should look like.